Makefile

```diff
@@ -426,15 +423,33 @@ INITIAL_MODULES = $(RUNTIME_CPP_COMPONENTS:%=$(BUILD_DIR)/initmod.%_32.o) $(RUNT
 .PHONY: all
 all: $(BIN_DIR)/libHalide.a $(BIN_DIR)/libHalide.so include/Halide.h include/HalideRuntime.h test_internal
 
+ifeq ($(USE_LLVM_SHARED_LIB), )
 $(BIN_DIR)/libHalide.a: $(OBJECTS) $(INITIAL_MODULES)
+	# Determine the relevant object files from llvm with a dummy
+	# compilation. Passing -t to the linker gets it to list which
+	# object files in which archives it uses to resolve
+	# symbols. We only care about the libLLVM ones.
+	@rm -rf $(BUILD_DIR)/llvm_objects
+	@mkdir -p $(BUILD_DIR)/llvm_objects
+	echo "int main(int, char **) {return 0;}" > $(BUILD_DIR)/empty_main.cpp
+	$(CXX) -o /dev/null $(BUILD_DIR)/empty_main.cpp $(OBJECTS) $(INITIAL_MODULES) -Wl,-t $(LLVM_STATIC_LIBS) -ldl -lz -lpthread | grep libLLVM | sed "s/[()]/ /g" > $(BUILD_DIR)/llvm_objects/list
```

Comment:
How about using the -shared flag, so no main is necessary? Not that the empty main is that big of a deal, but it would be ever so slightly simpler if it weren't needed.